src/testPane.ts

```diff
@@ -386,6 +392,7 @@ export function updateTestsForEnvironment(
         vcastEnviroList
       );
     }
+    controller.items.delete(enviroNode.id);
```

Comment:
We should add a comment about why this delete is required; reading the docs for `add` it _should not_ be required.

Reply:
Done.
I am planning to create an issue against vs code also

Reply:
I've already asked @Zbigor to try to create a reproducer for this.